cli/flox/src/utils/metrics.rs

```diff
@@ -44,6 +44,23 @@ macro_rules! subcommand_metric {
     }};
 }
 
+/// Creates a trace event for the given environment subcommand.
+///
+/// Do NOT inline functions as fields values because any child tracing events
+/// will cause this event to be logged at TRACE level for reasons unknown.
+///
+/// We set the target to `flox_command` so that we can filter for these exact events.
+#[macro_export]
+macro_rules! environment_subcommand_metric {
```

Comment:
Question blocking: Do we need a separate macro for this? I think you should be able to add this as a separate arm of the `subcommand_metric` macro. As it stands I'm a little confused when I would use one macro over the other.

Reply:
You would use `environment_subcommand_metric` when you have an EnvironmentSelect you can pass and want to include metrics for `--remote`. You would use `subcommand_metric` for all other metrics

To me it seems more readable to make it explicit that you want the environment metrics rather than making it implicit based on a match. If we just had `subcommand_metric` I'd probably copy and paste the first one I found when creating a new subcommand. Whereas if I saw we have both `subcommand_metric` and `environment_subcommand_metric` I'd probably think about it

Reply:
I'm suggesting a match arm that explicitly matches an `env = $env:expr` argument in the second position. I think this looks totally fine:
```
subcommand_metric("install", env = foo);
```

Reply:
Caveat that I'm not familiar with macro best practices. But if I glanced at that, I would think that `env` is treated the same as e.g. `search_term`. I would rather make it clear something's different with `environment_subcommand_metric` than making `env` behave magically

This
```
subcommand_metric("install", env = foo);
```
looks the same as
```
subcommand_metric!("search", search_term = &self.search_term);
```

Reply:
When I see `environment_subcommand_metric` and `subcommand_metric` exists I think "aren't these _all_ environment metrics?"

Reply:
Why would a search metric be an environment metric?

Reply:
Ok, all was an exaggeration, but _most_ of our metrics relate to environments. I'm not sure what the actual point is because it would still only be called from the `subcommand_metric` in either case though.

Reply:
My current understanding is that you're arguing for what is essentially function overloading, and I'm arguing it's not very readable. Since we're writing Rust which has decided function overloading is no bueno, I think its reasonable to follow the convention of no function overloading. As I'm unfamiliar with macros, my mind could totally be changed, but with my current understanding I think it makes sense to add a macro so I'm going to go ahead and merge